### PR TITLE
Update share behaviour

### PR DESCRIPTION
**Pull request by mishig25**
### TLDR: Change the current share conversation behaviour for desktop:

[internal slack link](https://huggingface.slack.com/archives/C052EKHD1U0/p1697718356502049?thread_ts=1697717286.065219&cid=C052EKHD1U0)

Currently: when you click `share conversation`, chat-ui opens [window.prompt](https://developer.mozilla.org/en-US/docs/Web/API/Window/prompt)

With this PR: just like [hf.co collection](https://huggingface.co/docs/hub/collections) sharing 
Screenshot 2023-12-19 at 3 11 59 PM

when you click `share conversation`, chat-ui saves the share url in user's clipboard and the share btn text becomes `Copied to clipboard` for 2 seconds
Screenshot 2023-12-19 at 3 11 36 PM

#### For mobile, nothing changes:

It shows the mobile share window as before

### CODE REVIEW

File: src/lib/components/chat/ChatWindow.svelte [hunk 1]
```diff
@@ -38,6 +40,7 @@
 
 	let loginModalOpen = false;
 	let message: string;
+	let timeout: ReturnType<typeof setTimeout>;
```
Comment (nsarrazin) [inline src/lib/components/chat/ChatWindow.svelte:44]:
```suggestion
	let isSharedRecently = false;
	$: $page.params.id && (isSharedRecently = false);
```

File: src/lib/components/chat/ChatWindow.svelte [hunk 2]
```diff
@@ -226,12 +246,19 @@
 				</p>
 				{#if messages.length}
 					<button
-						class="flex flex-none items-center hover:text-gray-400 hover:underline max-sm:rounded-lg max-sm:bg-gray-50 max-sm:px-2.5 dark:max-sm:bg-gray-800"
+						class="flex flex-none items-center hover:text-gray-400 max-sm:rounded-lg max-sm:bg-gray-50 max-sm:px-2.5 dark:max-sm:bg-gray-800"
 						type="button"
+						class:hover:underline={!$isConversationShared}
```
Comment (nsarrazin) [inline src/lib/components/chat/ChatWindow.svelte:251]:
```suggestion
						class:hover:underline={!isSharedRecently}
```

File: src/lib/components/chat/ChatWindow.svelte [hunk 3]
```diff
@@ -226,12 +246,19 @@
...
 						on:click={() => dispatch("share")}
+						disabled={$isConversationShared}
```
Comment (nsarrazin) [inline src/lib/components/chat/ChatWindow.svelte:253]:
```suggestion
						disabled={isSharedRecently}
```

File: src/lib/components/chat/ChatWindow.svelte [hunk 4]
```diff
@@ -226,12 +246,19 @@
 				</p>
 				{#if messages.length}
 					<button
-						class="flex flex-none items-center hover:text-gray-400 hover:underline max-sm:rounded-lg max-sm:bg-gray-50 max-sm:px-2.5 dark:max-sm:bg-gray-800"
+						class="flex flex-none items-center hover:text-gray-400 max-sm:rounded-lg max-sm:bg-gray-50 max-sm:px-2.5 dark:max-sm:bg-gray-800"
 						type="button"
+						class:hover:underline={!$isConversationShared}
 						on:click={() => dispatch("share")}
```
Comment (nsarrazin) [inline src/lib/components/chat/ChatWindow.svelte:252]:
```suggestion
						on:click={() => {
							dispatch("share");
							isSharedRecently = true;
							setTimeout(() => {
								isSharedRecently = false;
							}, 2000);
						}}
```

File: src/lib/components/chat/ChatWindow.svelte [hunk 5]
```diff
@@ -226,12 +246,19 @@
...
+						disabled={$isConversationShared}
 					>
-						<CarbonExport class="text-[.6rem] sm:mr-1.5 sm:text-primary-500" />
-						<div class="max-sm:hidden">Share this conversation</div>
+						{#if $isConversationShared}
```
Comment (nsarrazin) [inline src/lib/components/chat/ChatWindow.svelte:255]:
```suggestion
						{#if isSharedRecently}
```

**Review by nsarrazin — COMMENTED** [review]
I like the new behaviour, looks a lot better :grin: 

I think you can simplify it quite a bit and do it entirely in the `ChatWindow.svelte` without needing a store, see the suggestions, but maybe there's a reason to do it the other way, feel free to accept the changes or not!

File: src/lib/components/chat/ChatWindow.svelte (hunk 4 above)
Comment (nsarrazin) [reply src/lib/components/chat/ChatWindow.svelte:252]: this is the main piece, we dispatch the event and set the timeout here, shouldn't need much else

File: src/lib/components/chat/ChatWindow.svelte (hunk 5 above)
Comment (mishig25) [reply src/lib/components/chat/ChatWindow.svelte]: handled in https://github.com/huggingface/chat-ui/pull/645/commits/fdbb4cafbf846c95557cb7e2fe0ce45effcd4c76

File: src/lib/components/chat/ChatWindow.svelte (hunk 4 above)
Comment (mishig25) [reply src/lib/components/chat/ChatWindow.svelte]: handled in https://github.com/huggingface/chat-ui/pull/645/commits/fdbb4cafbf846c95557cb7e2fe0ce45effcd4c76

File: src/lib/components/chat/ChatWindow.svelte (hunk 3 above)
Comment (mishig25) [reply src/lib/components/chat/ChatWindow.svelte]: handled in https://github.com/huggingface/chat-ui/pull/645/commits/fdbb4cafbf846c95557cb7e2fe0ce45effcd4c76

File: src/lib/components/chat/ChatWindow.svelte (hunk 2 above)
Comment (mishig25) [reply src/lib/components/chat/ChatWindow.svelte]: handled in https://github.com/huggingface/chat-ui/pull/645/commits/fdbb4cafbf846c95557cb7e2fe0ce45effcd4c76

File: src/lib/components/chat/ChatWindow.svelte (hunk 1 above)
Comment (mishig25) [reply src/lib/components/chat/ChatWindow.svelte:45]: handled in https://github.com/huggingface/chat-ui/pull/645/commits/fdbb4cafbf846c95557cb7e2fe0ce45effcd4c76

**Comment by mishig25** [comment]
> I think you can simplify it quite a bit and do it entirely in the ChatWindow.svelte without needing a store, see the suggestions, but maybe there's a reason to do it the other way, feel free to accept the changes or not!

initially, I did the store thing to make sure to only show `Copied to clipboard` msg if [share function](https://github.com/huggingface/chat-ui/blob/update_share_btn/src/lib/utils/share.ts#L1-L2) succeeded. But I see that it overcomplicates the flow a bit and if there is an error in  [share function](https://github.com/huggingface/chat-ui/blob/update_share_btn/src/lib/utils/share.ts#L1-L2), it will in console.error anyways

**Review by gary149 — APPROVED** [review]

File: src/lib/components/chat/ChatWindow.svelte [hunk 6]
```diff
@@ -38,6 +39,9 @@
 
 	let loginModalOpen = false;
 	let message: string;
+	let timeout: ReturnType<typeof setTimeout>;
```
Comment (nsarrazin) [inline src/lib/components/chat/ChatWindow.svelte:42]: Kind of a nit or maybe I'm missing something: does the timeout needs to be referenced and handled in the `onDestroy` and can't we just do it like this?
```
on:click={() => {
	dispatch("share");
	isSharedRecently = true;
	setTimeout(() => {
		isSharedRecently = false;
	}, 2000);
}}
```

just curious, other than that lgtm

Comment (mishig25) [reply src/lib/components/chat/ChatWindow.svelte:42]: > does the timeout needs to be referenced and handled in the onDestroy and can't we just do it like this?

yes, for the edge case.

If a user clicks `share btn` & closes the window within less than 2 seconds, it would end up creating memory leak
